app/src/main/java/ua/dancecamp/vmedvediev/ua21summerdancecamp/UI/splash/SplashActivity.kt

```diff
@@ -31,40 +31,56 @@ class SplashActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_splash)
+
         splashViewModel.apply {
             getApplicationSettings()
             applicationSettings.observe(this@SplashActivity, Observer {
                 if (it != null) {
-                    val localeLanguage = baseContext.resources.configuration.locale.language
-                    val applicationLanguage = if (it.interfaceLanguage.isEmpty()) {
-                        localeLanguage
-                    } else {
-                        it.interfaceLanguage
-                    }
-                    val previousLocaleLanguage = if (it.localeLanguage.isEmpty()) {
-                        localeLanguage
-                    } else {
-                        it.localeLanguage
-                    }
-                    when {
-                        EventsCache.eventsList.isNotEmpty() && (previousLocaleLanguage == applicationLanguage) -> {
-                            startActivity(Router.prepareMainActivityIntent(this@SplashActivity))
-                            finish()
-                        }
-                        splashViewModel.getEvents().isNotEmpty() && (previousLocaleLanguage == applicationLanguage) -> {
-                            splashViewModel.setupLocalStorage()
-                            startActivity(Router.prepareMainActivityIntent(this@SplashActivity))
-                            finish()
-                        }
-                        else -> JsonParserAsyncTask(WeakReference(this@SplashActivity)).execute(applicationLanguage)
-                    }
+                    val applicationLanguage  = checkApplicationLanguage(it)
+                    val previousLocaleLanguage = checkPreviousLocaleLanguage(it)
+
+                    startMainActivity(previousLocaleLanguage, applicationLanguage)
+
                     updateLocale(applicationLanguage)
                     splashViewModel.saveApplicationSettigns(ApplicationSettings(it.id, applicationLanguage, applicationLanguage))
                 }
             })
         }
     }
 
+    private fun startMainActivity(previousLocaleLanguage: String, applicationLanguage: String) {
+        when {
+            EventsCache.eventsList.isNotEmpty() && (previousLocaleLanguage == applicationLanguage) -> {
+                startActivity(Router.prepareMainActivityIntent(this@SplashActivity))
+                finish()
+            }
+            splashViewModel.getEvents().isNotEmpty() && (previousLocaleLanguage == applicationLanguage) -> {
+                splashViewModel.setupLocalStorage()
+                startActivity(Router.prepareMainActivityIntent(this@SplashActivity))
+                finish()
+            }
+            else -> JsonParserAsyncTask(WeakReference(this@SplashActivity)).execute(applicationLanguage)
+        }
+    }
+
+    private fun checkApplicationLanguage(applicationSettings: ApplicationSettings) : String {
```

Comment:
Try to refactor these two functions: `checkApplicationLanguage()` and `checkApplicationLanguage()` bacause they look too similar